sharedbuffers/mapped_struct.py

```diff
@@ -1001,6 +1001,166 @@ def __str__(self):
+    def unpack_from(cls, buf, offs, idmap = None):
+        buf = _likerobuffer(buf)
+        if cython.compiled:
+            PyObject_GetBuffer(buf, cython.address(pybuf), PyBUF_SIMPLE)
+            pbuf = cython.cast(cython.p_uchar, pybuf.buf)
+            if offs >= pybuf.len:
+                PyBuffer_Release(cython.address(pybuf))
+                raise IndexError("Offset out of range")
+        else:
+            pbuf = buf
+        try:
+            if pbuf[offs] == 'm':
+                # inline bitmap
+                if cython.compiled and offs+7 >= pybuf.len:
+                    raise IndexError("Object spans beyond buffer end")
+                rv = []
+                for i in xrange(7):
+                    b = ord(pbuf[offs+1+i])
+                    if b:
+                        for j in xrange(8):
+                            if b & (1<<j):
+                                rv.append(i*8+j)
+                return frozenset(rv)
+            else:
+                return proxied_frozenset(proxied_list(buf, offs, idmap))
+        finally:
+            if cython.compiled:
+                if type(buf) is buffer:
+                    PyBuffer_Release(cython.address(pybuf))
+
+    def copy(self):
+        return self
+
+    def __len__(self):
+        return len(self.objlist)
+
+    @cython.locals(i1=cython.Py_ssize_t, i2=cython.Py_ssize_t,
+        cnt=cython.Py_ssize_t, step=cython.Py_ssize_t, lst='proxied_list')
+    def __contains__(self, elem):
+        i1 = 0
+        i2 = len(self.objlist)
+        cnt = i2 - i1
+        lst = self.objlist
+
+        while cnt > 0:
+            step = cnt >> 1
+            val = lst._getitem(step)
+            if val == elem:
+                return True
+            elif val < elem:
+                cnt -= step + 1
+            else:
+                cnt = step
+        return False
+
+    def __iter__(self):
+        return iter(self.objlist)
+
+    def union(self, *seqs):
+        return frozenset(self).union(*seqs)
+
+    def intersection(self, *seqs):
+        return frozenset(self).intersection(*seqs)
+
+    def difference(self, *seqs):
+        return frozenset(self).difference(*seqs)
+
+    def symmetric_difference(self, *seqs):
+        return frozenset(self).symmetric_difference(*seqs)
+
+    def __eq__(self, seq):
+        if not isinstance(seq, (set, frozenset, proxied_frozenset, mapped_frozenset)) or (
+              len(self) != len(seq)):
+            return False
+        for val in self:
+            if val not in seq:
+                return False
+        return True
+
+    @cython.ccall
+    @cython.locals(strict_subset=cython.bint)
+    def _subset(self, seq, strict_subset):
+        if len(self) > len(seq):
+            return False
+
+        for val in self:
+            if val not in seq:
+                return False
+        return not strict_subset or len(self) < len(seq)
+
+    def __lt__(self, seq):
+        return self._subset(seq, True)
```

Comment:
Same here. Self won't always be of the right type. You have to make your code support both calling conventions `(proxied_frozenset, something_else)` and `(something_else, proxied_frozenset)`.

Just test it, try: `frozenset([1]) < proxied_frozenset([1,2])`, you'll see it blows up.

Reply:
The test you proposed works fine for me ...

I can make the change, though. It's trivial.